src/Evolution/ComputeTags.hpp

```diff
@@ -11,81 +15,50 @@
 #include "Domain/Tags.hpp"
 #include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
 #include "Time/Tags.hpp"
+#include "Utilities/Gsl.hpp"
 #include "Utilities/TMPL.hpp"
 
-/// \cond
-namespace gsl {
-template <typename T>
-class not_null;
-}  // namespace gsl
-/// \endcond
-
-namespace evolution {
-namespace Tags {
+namespace evolution::Tags {
 /*!
- * \brief Use the `AnalyticSolutionTag` to compute the analytic solution of the
- * tags in `AnalyticFieldsTagList`.
+ * \brief Computes the analytic solution and adds `::Tags::Analytic` of the
+ * `std::optional<Tensor>`s to the DataBox.
  */
-template <size_t Dim, typename AnalyticSolutionTag,
-          typename AnalyticFieldsTagList>
-struct AnalyticCompute : ::Tags::AnalyticSolutions<AnalyticFieldsTagList>,
-                         db::ComputeTag {
+template <size_t Dim, typename AnalyticFieldsTagList>
+struct AnalyticSolutionsCompute
+    : ::Tags::AnalyticSolutions<AnalyticFieldsTagList>,
+      db::ComputeTag {
+  using field_tags = AnalyticFieldsTagList;
   using base = ::Tags::AnalyticSolutions<AnalyticFieldsTagList>;
   using return_type = typename base::type;
   using argument_tags =
-      tmpl::list<AnalyticSolutionTag,
+      tmpl::list<::Tags::AnalyticSolutionOrData,
                  domain::Tags::Coordinates<Dim, Frame::Inertial>, ::Tags::Time>;
+  template <typename AnalyticSolution>
   static void function(
       const gsl::not_null<return_type*> analytic_solution,
-      const typename AnalyticSolutionTag::type& analytic_solution_computer,
+      const AnalyticSolution& analytic_solution_computer,
       const tnsr::I<DataVector, Dim, Frame::Inertial>& inertial_coords,
       const double time) {
-    *analytic_solution =
-        variables_from_tagged_tuple(analytic_solution_computer.variables(
-            inertial_coords, time, AnalyticFieldsTagList{}));
+    if constexpr (std::is_convertible_v<AnalyticSolution*,
+                                        MarkAsAnalyticSolution*>) {
+      *analytic_solution =
+          variables_from_tagged_tuple(analytic_solution_computer.variables(
+              inertial_coords, time, AnalyticFieldsTagList{}));
+    } else {
+      (void)analytic_solution_computer;
+      (void)inertial_coords;
+      (void)time;
```

Comment:
(optional) I prefer `[[maybe_unused]]` these days

Reply:
I don't like it in these cases because I prefer to explicitly acknowledge which arguments I'm not using in the branch